core/unit_test/cuda/TestCuda_TeamScratch.cpp

```diff
@@ -47,22 +47,121 @@
 
 namespace Test {
 
+namespace Impl {
+
+struct CudaStreamScratchTestFunctor {
+  using team_t    = Kokkos::TeamPolicy<Kokkos::Cuda>::member_type;
+  using scratch_t = Kokkos::View<int64_t*, Kokkos::Cuda::scratch_memory_space>;
+
+  Kokkos::View<int64_t, Kokkos::CudaSpace, Kokkos::MemoryTraits<Kokkos::Atomic>>
+      counter;
+  int N, M;
+  CudaStreamScratchTestFunctor(
+      Kokkos::View<int64_t, Kokkos::CudaSpace> counter_, int N_, int M_)
+      : counter(counter_), N(N_), M(M_) {}
+
+  KOKKOS_FUNCTION
+  void operator()(const team_t& team) const {
+    scratch_t scr(team.team_scratch(1), M);
+    Kokkos::parallel_for(Kokkos::TeamThreadRange(team, 0, M),
+                         [&](int i) { scr[i] = 0; });
+    team.team_barrier();
+    for (int i = 0; i < N; i++) {
+      Kokkos::parallel_for(Kokkos::TeamThreadRange(team, 0, M),
+                           [&](int j) { scr[j] += 1; });
+    }
+    team.team_barrier();
+    Kokkos::parallel_for(Kokkos::TeamThreadRange(team, 0, M), [&](int i) {
+      if (scr[i] != N) counter()++;
+    });
+  }
+};
+
+void cuda_stream_scratch_test_one(
+    int N, int T, int M_base, Kokkos::View<int64_t, Kokkos::CudaSpace> counter,
+    Kokkos::Cuda cuda, int tid) {
+  int M = M_base + tid * 5;
+  Kokkos::TeamPolicy<Kokkos::Cuda> p(cuda, T, 64);
+  using team_t    = Kokkos::TeamPolicy<Kokkos::Cuda>::member_type;
+  using scratch_t = Kokkos::View<int64_t*, Kokkos::Cuda::scratch_memory_space>;
+
+  int bytes = scratch_t::shmem_size(M);
+
+  for (int r = 0; r < 15; r++) {
+    Kokkos::parallel_for("Run", p.set_scratch_size(1, Kokkos::PerTeam(bytes)),
+                         CudaStreamScratchTestFunctor(counter, N, M));
+  }
+}
+
+void cuda_stream_scratch_test(
+    int N, int T, int M_base,
+    Kokkos::View<int64_t, Kokkos::CudaSpace> counter) {
+  int K = 4;
+  cudaStream_t stream[4];
+  Kokkos::Cuda cuda[4];
+  for (int i = 0; i < K; i++) {
+    cudaStreamCreate(&stream[i]);
+    cuda[i] = Kokkos::Cuda(stream[i]);
+  }
+  // Test that growing scratch size in subsequent calls doesn' crash things
```

Comment:
```suggestion
  // Test that growing scratch size in subsequent calls doesn't crash things
```

Reply:
Does somebody else want to unroll all the changes and redo the commits to add the t? I don't have time right now. If nobody else feels the urgent need I will ignore this :-)